### PR TITLE
[Bug 18564] Support for the remote debugger library

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

**Comment by montegoulding** [comment]
https://github.com/livecode/livecode-private/pull/109 needs to be merged before this PR

File: Toolset/home.livecodescript [hunk 1]
```diff
@@ -1179,6 +1179,10 @@ command revInternal__InitialiseLibraries
    # AL-2015-09-02: [[ Bug 15841 ]] Temporarily add project browser library to prevent preference errors
    insert script of stack "revIDEProjectBrowserLibrary" into back
    
+   if the editionType is "professional" then
+      revInternal__LoadLibrary "com.livecode.script-library.remotedebugger"
```
Comment (montegoulding) [inline Toolset/home.livecodescript]: @livecodeali Any thoughts on putting libraries somewhere where they can be automatically loaded? Maybe they should go into the Extensions dir?

**Comment by livecodeali** [comment]
For future reference, this is the change to the error template script
https://gist.github.com/livecodeali/28d87302e351ea1095d12f93c20144fc

**Comment by livecodeali** [comment]
@montegoulding This looks fine to me. I'm just wondering about all the checks of `the environment contains "development"` - do you think they should just be `... is "development"`? As the debugger isn't particularly relevant in `development command line` environment.

**Comment by montegoulding** [comment]
@livecodeali when I was implementing it I couldn't help wondering if it could be used for running tests on mobile so I did that for future proofing but it needs more thought... it may be that you still wouldn't use development command line for that anyway...

**Comment by livecodeali** [comment]
@montegoulding Hmm, you would possibly have to if the receiving end relies on IDE infrastructure. Either way, I don't see a problem with leaving it in just now.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 71d9ee9

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 71d9ee9108d474a5164259edad41b19decd4509e

**Comment by montegoulding** [comment]
OK, cheers. BTW this shouldn't be merged until the remote debugger is because it references it in the home stack script.

**Comment by peter-b** [comment]
This all looks fairly reasonable to me!
@livecode-vulcan review ok 7f91002

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 7f910025971ec04ec973a5968a2f660526692502

**Comment by livecodepanos** [comment]
I set this to WIP to prevent accidental merging, until the counterpart in the private repo is merged

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 33efbdb

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 33efbdbe5c2bcf8cdbb329907e8e478fcca7defd

**Pull request merged**